backend/src/apiserver/template/v2_template.go

```diff
@@ -77,9 +77,20 @@ func (t *V2Spec) ScheduledWorkflow(modelJob *model.Job) (*scheduledworkflow.Sche
 		}
 	}
 
+	var pipeline_options argocompiler.Options
+	for _, platform := range t.platformSpec.Platforms {
+		if platform.PipelineConfig.SemaphoreKey != "" || platform.PipelineConfig.MutexName != "" {
+			pipeline_options = argocompiler.Options{
+				SemaphoreKey: platform.PipelineConfig.SemaphoreKey,
+				MutexName:    platform.PipelineConfig.MutexName,
```

Comment:
Let's add these individually, and only if they are specified in the IR. Users will typically only add one or the other. (So don't add mutex if the user only wants a semaphore, and vice versa.)

Reply:
Done, added these individually.